Flickr/Flickr/ImageCollectionViewController.swift

```diff
@@ -10,17 +10,134 @@
 import UIKit
 import Alamofire
 
-class ImageCollectionViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource, UITextFieldDelegate {
-  
-    fileprivate var imageData: [Photo] = [Photo]()
+class ImageCollectionViewController: UIViewController,  UICollectionViewDelegate, UICollectionViewDataSource, UITextFieldDelegate, UICollectionViewDelegateFlowLayout, UIScrollViewDelegate, UITableViewDelegate, UITableViewDataSource, recentTableCellDelegate {
+    
+    var pageFlickr = 1
+    var recentIndexCell: Int?
+    var searchActive : Bool = false
+    var actualPosition: CGPoint?
+    var searchList = [String]()
+    var popularImageData: [Photo] = [Photo]()
+    var popularImageSizes: [CGSize] = [CGSize]()
```

Comment:
make thematic grouping of this variables for better navigation
common variables
all search variables
all popular variables
outlets

could make one space between them